keras_cv/layers/object_detection/retina_net_label_encoder.py

```diff
@@ -56,12 +64,19 @@ def __init__(
         self.matched_boxes_metric = tf.keras.metrics.BinaryAccuracy(
             name="percent_boxes_matched_with_anchor"
         )
+        self.positive_threshold = positive_threshold
+        self.negative_threshold = negative_threshold
+        self.box_matcher = box_matcher.ArgmaxBoxMatcher(
+            thresholds=[negative_threshold, positive_threshold],
+            match_values=[-1, -2, 1],
+            force_match_for_each_col=False,
+        )
         self.built = True
 
-    def _match_anchor_boxes(
-        self, anchor_boxes, gt_boxes, match_iou=0.5, ignore_iou=0.4
-    ):
-        """Matches ground truth boxes to anchor boxes based on IOU.
+    def _encode_sample(self, gt_boxes, gt_classes, anchor_boxes):
+        """Creates box and classification targets for a single sample
```

Comment:
is this still "for a single sample" if the implementation is batched now?  Lets updated

Reply:
Done